plugins/woocommerce/client/admin/webpack.config.js

```diff
@@ -182,7 +182,31 @@ const webpackConfig = {
 		! process.env.STORYBOOK && new ForkTsCheckerWebpackPlugin(),
 		! process.env.STORYBOOK &&
 			new TypeScriptWarnOnlyWebpackPlugin( [
-				/* specify TS error codes here, e.g. 'TS2322' */
+				// these are the errors that have been converted into warnings during the react-18 upgrade
+				// the number is the original number of instances of that error when this comment is written
+				// hopefully this information helps prioritize fixing the errors :)
+				'TS2349', // ~107: This expression is not callable.
+				'TS2578', // ~69: Unused '@ts-expect-error' directive.
+				'TS2554', // ~60: Expected 2 arguments, but got 1.
+				'TS2339', // ~51: Property 'getActivePlugins' does not exist on type 'never'.
+				'TS7006', // ~38: Parameter implicitly has an 'any' type.
+				'TS2322', // ~31: Type mismatch in assignment.
+				'TS2503', // ~27: Cannot find namespace 'Slot'.
+				'TS2559', // ~11: Type mismatch in object property.
+				'TS2786', // ~10: Component cannot be used as a JSX element.
+				'TS2604', // ~10: JSX element type has no construct or call signatures.
+				'TS2345', // ~7: Argument type mismatch.
+				'TS2741', // ~5: Missing required property in type.
+				'TS2305', // ~4: Module has no exported member.
+				'TS2347', // ~3: Untyped function calls may not accept type arguments.
+				'TS2769', // ~2: No overload matches this call.
+				'TS2558', // ~2: Unexpected number of type arguments.
+				'TS2344', // ~2: Type does not satisfy a specific constraint.
+				'TS18046', // ~2: Variable is of type 'unknown'.
+				'TS2677', // 1: Type predicate's type must be assignable to its parameter's type.
+				'TS2571', // 1: Object is of type 'unknown'.
+				'TS2352', // 1: Invalid type conversion.
+				'TS18049', // 1: Nullable or undefined value encountered.
```

Comment:
To confirm, these errors are the ones that appeared as a result of PR?

Reply:
I don't see TS2352, TS2604 & TS2677 errors when running `npx tsc --noEmit` on this branch

Reply:
> To confirm, these errors are the ones that appeared as a result of PR?

Yep, since running `npx tsc` on trunk doesn't report anything

> I don't see TS2352, TS2604 & TS2677 errors when running npx tsc --noEmit on this branch

It's varied a little since I run those numbers about a week+ ago with changes on top of that both from the PR and trunk. Here's the latest output I have:

command: `npx tsc | awk '/error TS[0-9]+:/ {sub(/^.*error /, ""); split($0, arr, ": "); count[arr[1]]++; desc[arr[1]] = arr[2]} END {for (e in count) printf "%d %s: %s\n", count[e], e, desc[e]}' | sort -nr`

```
107 TS2349: This expression is not callable.
70 TS2578: Unused '@ts-expect-error' directive.
62 TS2554: Expected 2 arguments, but got 1.
58 TS2339: Property 'getActivePlugins' does not exist on type 'never'.
41 TS7006: Parameter 'fills' implicitly has an 'any' type.
31 TS2322: Type '{ id
25 TS2503: Cannot find namespace 'Slot'.
11 TS2559: Type '{ children
10 TS2786: 'ConnectForm' cannot be used as a JSX component.
10 TS2604: JSX element type 'ConnectForm' does not have any construct or call signatures.
7 TS2345: Argument of type 'number | undefined' is not assignable to parameter of type 'string'.
6 TS2741: Property 'label' is missing in type '{ checked
4 TS2347: Untyped function calls may not accept type arguments.
4 TS2305: Module '"@wordpress/notices"' has no exported member 'Options'.
2 TS2769: No overload matches this call.
2 TS2558: Expected 0 type arguments, but got 1.
2 TS2344: Type 'Product | ProductVariation' does not satisfy the constraint 'StoreDescriptor<any> | MapSelect'.
2 TS18046: 'a' is of type 'unknown'.
1 TS2677: A type predicate's type must be assignable to its parameter's type.
1 TS2571: Object is of type 'unknown'.
1 TS2352: Conversion of type '{ capabilities
1 TS18049: 'errorInfo.componentStack' is possibly 'null' or 'undefined'.
```

I still see TS2352 and TS2604, but not TS2677

The descriptions are a bit different because I ran the previous one through chatgpt to get generalised descriptions rather than truncated ones

Reply:
I wasn't running the command on all the repo areas, that's why I wasn't seeing those codes

Reply:
@rjchow 
> I was running the command in ./plugins/woocommerce/client/admin - we're commenting on the webpack config that builds admin 😅

Double 😅😅 on my end, this was an obvious oversight on my part due to being stretched a bit thin

> I'm happy to take on the work of assigning/delegating for JS packages and admin, if you'd like to focus on woocommerce-blocks?

I can take care of the assigning/delegating. 

> Also, do we need to enable TS checking on woocommerce-blocks ? It doesn't seem to be part of the webpack config or pnpm build

Let's not do that now, I'll need to see when this was disabled, and I think we'd have to fix that outside of this project, as the blocks are already on React 18. I'll take care of documenting and delegation these as well.